agent/task.go

```diff
@@ -242,6 +249,24 @@ func (tc *taskContext) hadTimedOut() bool {
 	return tc.timedOut()
 }
 
+func (tc *taskContext) getOomTrackerInfo() apimodels.OOMTrackerInfo {
+	tc.RLock()
```

Comment:
Could you explain the reason to put this function and the ntext behind a mutex?

Reply:
My thinking was that the fields of the checker are written/read from separate threads.
On the other hand, the main thread will not read this until the first exits (because we [wait](https://github.com/evergreen-ci/evergreen/blob/2c4eeec32698f70c87c4153f52a528ef58a4f1a5/agent/agent.go#L399)), so I'll drop it.

Reply:
actually, they're both in the same thread. I'll definitely remove it

Reply:
removed